kimchi/src/tests/foreign_field_mul.rs

```diff
@@ -1407,3 +1321,41 @@ fn test_rand_foreign_field_element_with_bound_overflows_7() {
     )
     .expect("Failed to get element with bound overflow");
 }
+
+#[test]
+fn test_native_modulus_constraint() {
```

Comment:
Really cool that you found this counter example, really well done! I will look at the details of it to see if we can come up with an "explainer" of this circumstance.

Reply:
Awesome - thanks!